tests/types/tillegaltyperecursion.nim

```diff
@@ -1,66 +0,0 @@
-discard """
-  cmd: "nim $target --threads:on $options $file"
-  errormsg: "illegal recursion in type 'TIRC'"
-  line: 16
-"""
-
-import events
-import net
-import strutils
-import os
-
```

Comment:
keep this test but simplify it as much as possible (and remove reference to events etc); it should test that illegal type recursion leads to a clean error